mreg/models/auth.py

```diff
@@ -10,3 +38,53 @@ def group_list(self):
         if self._group_list is None:
             self._group_list = list(self.groups.values_list("name", flat=True))
         return self._group_list
+
+    @property
+    def is_mreg_superuser(self):
+        return self.is_member_of_any(MregAdminGroup.SUPERUSER.settings_groups_or_raise())
+
+    @property
+    def is_mreg_admin(self):
+        return self.is_member_of_any(MregAdminGroup.ADMINUSER.settings_groups_or_raise())
+
+    @property
+    def is_mreg_group_admin(self):
+        return self.is_member_of_any(MregAdminGroup.GROUP_ADMIN.settings_groups_or_raise())
+
+    @property
+    def is_mreg_network_admin(self):
+        return self.is_member_of_any(MregAdminGroup.NETWORK_ADMIN.settings_groups_or_raise())
+    
+    @property
+    def is_mreg_dns_wildcard_admin(self):
+        return self.is_member_of_any(MregAdminGroup.DNS_WILDCARD.settings_groups_or_raise())
+    
+    @property
+    def is_mreg_dns_underscore_admin(self):
+        return self.is_member_of_any(MregAdminGroup.DNS_UNDERSCORE.settings_groups_or_raise())
+
+    @property
+    def is_mreg_hostpolicy_admin(self):
+        return self.is_member_of_any(MregAdminGroup.HOSTPOLICY_ADMIN.settings_groups_or_raise())
+
+    @property
+    def is_mreg_superuser_or_admin(self):
+        return self.is_mreg_admin or self.is_mreg_superuser
+
+    @property
+    def is_mreg_superuser_or_hostpolicy_admin(self):
+        return self.is_mreg_hostpolicy_admin or self.is_mreg_superuser
+
+    @classmethod
+    def from_request(cls, request: HttpRequest) -> 'User':
+        if not request.user.is_authenticated:
+            raise exceptions.NotAuthenticated("Attempted to coerce an authenticated user from unauthenticated request")
```

Comment:
In order to not leak internal implementation details, could we instead log that message and raise a more generic 401 message?

Reply:
Yep. Done.